notebooks/203-meter-reader/203-meter-reader-digital.ipynb

```diff
@@ -0,0 +1,1046 @@
+    "IMG_FILE_NAME = IMG_URL.split(\"/\")[-1]\n",
+    "utils.download_file(IMG_URL, show_progress=False)\n",
+    "\n",
+    "# The coordinates of the corners of the screen\n",
+    "POINTS = [[275, 676],    # Left top, [x, y]\n",
+    "          [1775, 704],   # right top\n",
+    "          [1829, 2439],  # right bottom\n",
+    "          [205, 2469]]   # left bottom\n",
+    "\n",
+    "# The size of the screen\n",
+    "DESIGN_SHAPE = (1500, 1800)\n",
+    "\n",
+    "# Output template\n",
+    "RESULT_TEMP = {\"Value\":\"\"}\n",
+    "\n",
+    "# features and the layout information\n",
+    "DESIGN_LAYOUT = {'Value':[470, 436, 1050, 592]}\n",
+    "\n",
+    "# a special pre-processing for the LCD screen\n",
+    "def pre_processing_for_LCD(cut_img):\n",
+    "    # BGR-image to BINARY-image\n",
+    "    cut_img = cv2.cvtColor(cut_img, cv2.COLOR_BGR2GRAY)\n",
+    "    _, cut_img = cv2.threshold(cut_img, 64, 255, cv2.THRESH_BINARY)\n",
+    "    \n",
+    "    # delete the area without text\n",
+    "    step_size = 30\n",
+    "    for i in range(0, cut_img.shape[1], step_size):\n",
+    "        rate = len(np.where(cut_img[:, i:i + step_size] > 128)[0]) / (cut_img[:, i:i + step_size].shape[0] * cut_img[:, i:i + step_size].shape[1])\n",
+    "        if rate <= 0.95:\n",
+    "            cut_img = cut_img[:, i:]\n",
+    "            break\n",
+    "    \n",
+    "    # [h, w] to [h, w, 3]\n",
+    "    cut_img = np.expand_dims(cut_img, axis=2)\n",
+    "    cut_img = np.concatenate((cut_img, cut_img, cut_img), axis=-1)\n",
+    "    \n",
+    "    return cut_img\n",
+    "    \n",
+    "# Congiguration for postprocessing of the results\n",
+    "RESULT_POST = {\"Value\":[['RP', ' ', ''], ['AD', -2]]}  # add point"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ec9789d1-465f-41b9-b1d4-8e52cb81f906",
+   "metadata": {},
+   "source": [
+    "### Pre-processing"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "28d164fa-59ec-4aa8-ba2f-8e270e762ddb",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# read image\n",
+    "img = cv2.imread(IMG_FILE_NAME)\n",
+    "\n",
+    "# pre-processing\n",
+    "img = pre_processing(img, POINTS, DESIGN_SHAPE)\n",
+    "\n",
+    "# Show pre-processed image\n",
+    "plt.imshow(img)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8a958dfe-2aa7-4d64-a974-21522e0d86dc",
+   "metadata": {},
+   "source": [
+    "### Recognition"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "92b43f6b-fb1b-4cb4-a9cd-a6f7bf1812fc",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# the input for recognition need the image, DESIGN information, compiled_model\n",
+    "def main_for_electricity_meter(img, DESIGN_LAYOUT, RESULT_TEMP, rec_compiled_model, rec_output_layer, text_decoder):\n",
```

Comment:
Could you help to add some comments on these key functions to explain the meaning of each parameter ? For example:

  """

  Batch for text recognition

  Parameters:

    img_crop_list: processed detected bounding box images 

    img_num: number of bounding boxes from text detection

    indices: sorting for bounding boxes to speed up text recognition

    beg_img_no: the beginning number of bounding boxes for each batch of text recognition inference

    batch_num: number of images for each batch

  """